app/actions/remote/team.ts

```diff
@@ -288,6 +289,19 @@ export async function handleTeamChange(serverUrl: string, teamId: string) {
         models.push(...history);
     }
 
+    // If the team is constrained, fetch it's groups
+    const team = await getTeamById(operator.database, teamId);
+    if (team && team.isGroupConstrained) {
+        const {groups, groupTeams} = await fetchGroupsForTeam(serverUrl, teamId, true);
```

Comment:
Do we need to await here? What is the impact of this delaying the switch? And what happens if I switch repeatedly over and over again?

Reply:
Changed to be async at the end.